spec/system/admin/order_spec.rb

```diff
@@ -179,9 +179,9 @@ def new_order_with_distribution(distributor, order_cycle)
     visit spree.edit_admin_order_path(order)
 
     select2_select product.name, from: 'add_variant_id', search: true
-
     find('button.add_variant').click
     # Wait for JS
+    sleep(1)
     page.has_selector?("table.index tbody tr td")
```

Comment:
I think if we change this line to an `expect` then the `sleep` might not be required...? :thinking:

```diff
-- page.has_selector?("table.index tbody tr td")
++ expect(page).to have_selector("table.index tbody tr td")
```

Reply:
Hey @Matt-Yorkley , thanks - I'd think the same...

I've tried this locally but get 5 pass out of 10, with:
`expected to find visible css "td" with text "Product #1 - 7326" but there were no matches`

Reply:
`has_selector?` waits as well. There's no difference in flakiness. I'm more wondering if we need that statement if it doesn't actually work.

This kind of expectation has caused flakiness in the past. I suspect a bug in Capybara. From memory, it's the combination of a selector with text. It doesn't recognise the text change on an existing element. Or do we replace the element? I don't know. Hopefully these edge-cases will disappear with removing AngularJS.